### ISSUE TITLE
Line/Col Mapping Performance Issues

### ISSUE THREAD

**Issue by dtg3** [iTrace-Dev/iTrace-VisualStudio#23]
### This needs to be verified on higher end hardware ###
The line/col mapping code used by the VS plugin is having a large performance impact on the plugin. It appears that it cannot, and has never, actually been able to keep up with the data transmitted by the core.

While a solution at this time is unknown possible avenues might be:
* multi threading
* throttling the data rate
* a different line/col lookup solution altogether

Any and all input is welcome on this issue.

**Comment by dtg3** [comment]
Socket timeout solution option has been removed as that was NOT a feature of QTCPSocket. Packet drops were due to the VS plugin using a timer that would read from a static member field that held data from the socket. Since the timer event was arbitrary with respect to receiving new data, this sometimes yielded duplicate data points being processed and missed data points.

**Comment by dtg3** [comment]
Tested on higher end hardware (i7 3.2Ghz, 16GB Ram, SSD).
Approximate run time in milliseconds for the line/col look up function ONLY is:
Mean: 35 ms
Mode: 27 ms
Median: 31 ms
Min: 13 ms
Max: 124 ms

To compare, data speeds for a 60Hz tracker is about one data point every 17 ms and for 160Hz every 6ms.

Attempts at multi-threading the line/col look up has been unsuccessful so far due to crashes from within that function (it was only ever intended to be run on the GUI thread). Other options are being tested, but if anyone uncovers some useful alternative method for line/col mapping or has ideas for optimizations, I'm open to all suggestions.

For right now, a buffered input and output version exists in the new_design branch. I'm not dead set on the new design being the best option for this either, so suggestions there are also welcome.

**Comment by dtg3** [comment]
An alternative solution for line/col mapping has been implemented in the `new_design` branch.

Current performance when using the mouse tracker at roughly 60Hz seems to keep up well with the data flow. In terms of dropped points, the core recorded 2,146 gazes and the VS plugin processed 2,062 (loss of 84 points). The dropped points can occur when editor windows are switched/opened/closed. Since the stimuli is in flux anyway, this seems like a reasonable amount of data loss.

In regards to incorrect window mapping, it is possible as new windows are switched/opened/closed this could happen. On my laptop I've noticed a few points (around 3) that could be mismatches attributed to a editor switch.

Additional testing with a tracker is required, but the current approach seems viable for optimal data processing throughput.
